mpt_trie/src/lib.rs

```diff
@@ -17,10 +17,11 @@
 
 pub mod nibbles;
 pub mod partial_trie;
+pub mod special_query;
 mod trie_hashing;
 pub mod trie_ops;
 pub mod trie_subsets;
-mod utils;
+pub mod utils;
```

Comment:
Maybe I should move the `pub` things outside of `utils.rs`, but right now `utils` needs to be pub in order to access the public types/functions (and public function in the `impl` blocks, although I'm not sure).

I was a bit unsure if I should have added this public logic to `utils`, but I didn't see a good place to put them otherwise. It's a bunch of assorted logic that is needed by users of the library.